test/testsuites/packagestests.go

```diff
@@ -102,7 +107,139 @@ func testOpSrcWithConflictingPackages(t *testing.T) {
+	namespace, err := test.NewTestCtx(t).GetNamespace()
+	assert.NoError(t, err)
+
+	packages := "camel-k-marketplace-e2e-tests-k,missing-package"
+	err = runSourceTest(namespace,
+		"",
+		packages,
+		"Configuring",
+		fmt.Sprintf("Unable to resolve the source - no source contains the requested package(s) [%s]", strings.Replace(packages, ",", " ", -1)),
+	)
+	assert.NoError(t, err)
+}
+
+// nonExistingSource ensures that if the provided source does not exist the CatalogSourceConfig
+// will be placed in the Configuring phase with the expected messages.
+func nonExistingSource(t *testing.T) {
+	namespace, err := test.NewTestCtx(t).GetNamespace()
+	assert.NoError(t, err)
+
+	source := "bad-source"
+	err = runSourceTest(namespace,
+		source,
+		"camel-k-marketplace-e2e-tests",
+		"Configuring",
+		fmt.Sprintf("Provided source (%s) does not exist", source),
+	)
+	assert.NoError(t, err)
+}
+
+// sourceMissingPackages ensures that if the provided source does not contain the expected
+// packages the CatalogSourceConfig will be placed in the Configuring phase with the expected
+// messages.
+func sourceMissingPackages(t *testing.T) {
+	namespace, err := test.NewTestCtx(t).GetNamespace()
+	assert.NoError(t, err)
+
+	err = runSourceTest(namespace,
+		"test-operators",
+		"camel-k-marketplace-e2e-tests,missing-package",
+		"Configuring",
+		"Still resolving package(s) - missing-package. Please make sure these are valid packages within the test-operators OperatorSource.",
+	)
+	assert.NoError(t, err)
+}
+
+func runSourceTest(namespace, source, packages, expectedPhase, expectedMessage string) error {
+	// Get global framework variables.
+	client := test.Global.Client
+
+	// Create a new CatalogSourceConfig with a non-existing targetNamespace.
+	csc := &v2.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: v2.CatalogSourceConfigKind,
+		}, ObjectMeta: metav1.ObjectMeta{
+			Name:      cscName,
+			Namespace: namespace,
+		},
+		Spec: v2.CatalogSourceConfigSpec{
+			Source:          source,
+			TargetNamespace: "openshift-marketplace",
+			Packages:        packages,
+		}}
+
+	// Create the CatalogSourceConfig and if an error occurs do not run tests that
+	// rely on the existence of the CatalogSourceConfig.
+	// The CatalogSourceConfig is created with nil ctx and must be deleted manually before test suite exits.
+	err := helpers.CreateRuntimeObject(client, nil, csc)
+	if err != nil {
+		return err
+	}
+
+	// Check that the CatalogSourceConfig with an non-existing targetNamespace eventually reaches the
+	// configuring phase with the expected message.
+	err = helpers.WaitForExpectedPhaseAndMessage(test.Global.Client, cscName, namespace, expectedPhase, expectedMessage)
+	if err != nil {
+		return err
+	}
+
+	// Delete the CatalogSourceConfig.
+	err = helpers.DeleteRuntimeObject(client, csc)
+	if err != nil {
+		return err
+	}
+
+	// Wait for the CatalogSourceConfig to be deleted.
+	err = helpers.WaitForNotFound(client, csc, namespace, cscName)
```

Comment:
Nit: this can just be `return helpers.WaitForNotFound(client, csc, namespace, cscName)`

Reply:
Good point